packages/@aws-cdk/aws-apigatewayv2/lib/common/api-mapping.ts

```diff
@@ -82,6 +82,11 @@ export class ApiMapping extends Resource implements IApiMapping {
    */
   public readonly mappingKey?: string;
 
+  /**
+   * API domain name
+   */
+  public readonly domainName?: string;
```

Comment:
This should not be optional.

Why not just set the type to `IDomainName` instead?

Reply:
Good idea, I've updated the type here